CovertInfiltration/Src/CovertInfiltration/Classes/X2Ability_InfiltrationAbilitySet.uc

```diff
@@ -0,0 +1,132 @@
+
+	Template.AddTargetEffect(PersistentStatChangeEffect);
+
+	Template.BuildNewGameStateFn = TypicalAbility_BuildGameState;
+
+	return Template;	
+}
+
+static function X2AbilityTemplate AdventDisguiseStats()
+{
+	local X2AbilityTemplate                 Template;	
+	local X2AbilityTrigger					Trigger;
+	local X2AbilityTarget_Self				TargetStyle;
+	local X2Effect_PersistentStatChange		PersistentStatChangeEffect;
+
+	`CREATE_X2ABILITY_TEMPLATE(Template, 'AdventDisguiseStats');
+
+	Template.AbilitySourceName = 'eAbilitySource_Item';
+	Template.eAbilityIconBehaviorHUD = EAbilityIconBehavior_NeverShow;
+	Template.Hostility = eHostility_Neutral;
+	Template.bDisplayInUITacticalText = false;
+	
+	Template.AbilityToHitCalc = default.DeadEye;
+	
+	TargetStyle = new class'X2AbilityTarget_Self';
+	Template.AbilityTargetStyle = TargetStyle;
+
+	Trigger = new class'X2AbilityTrigger_UnitPostBeginPlay';
+	Template.AbilityTriggers.AddItem(Trigger);
+
+	PersistentStatChangeEffect = new class'X2Effect_PersistentStatChange';
+	PersistentStatChangeEffect.BuildPersistentEffect(1, true, false, false);
+	PersistentStatChangeEffect.AddPersistentStatChange(eStat_HP, default.ADVENT_DISGUISE_HEALTH_BONUS);
+	PersistentStatChangeEffect.AddPersistentStatChange(eStat_Mobility, default.ADVENT_DISGUISE_MOBILITY_BONUS);
+	PersistentStatChangeEffect.AddPersistentStatChange(eStat_DetectionModifier, default.ADVENT_DISGUISE_DETECTION_MODIFIER);
+	Template.AddTargetEffect(PersistentStatChangeEffect);
+
+	Template.BuildNewGameStateFn = TypicalAbility_BuildGameState;
+
+	return Template;	
+}
+
+static function X2AbilityTemplate HolographicDisguiseStats()
+{
+	local X2AbilityTemplate                 Template;
+	local X2AbilityTrigger					Trigger;
+	local X2AbilityTarget_Self				TargetStyle;
+	local X2Effect_PersistentStatChange		PersistentStatChangeEffect;
+
+	`CREATE_X2ABILITY_TEMPLATE(Template, 'HolographicDisguiseStats');
+
+	Template.AbilitySourceName = 'eAbilitySource_Item';
+	Template.eAbilityIconBehaviorHUD = EAbilityIconBehavior_NeverShow;
+	Template.Hostility = eHostility_Neutral;
+	Template.bDisplayInUITacticalText = false;
+
+	Template.AbilityToHitCalc = default.DeadEye;
+
+	TargetStyle = new class'X2AbilityTarget_Self';
+	Template.AbilityTargetStyle = TargetStyle;
+
+	Trigger = new class'X2AbilityTrigger_UnitPostBeginPlay';
+	Template.AbilityTriggers.AddItem(Trigger);
+
+	PersistentStatChangeEffect = new class'X2Effect_PersistentStatChange';
+	PersistentStatChangeEffect.BuildPersistentEffect(1, true, false, false);
+	PersistentStatChangeEffect.AddPersistentStatChange(eStat_HP, default.HOLOGRAPHIC_DISGUISE_HEALTH_BONUS);
+	PersistentStatChangeEffect.AddPersistentStatChange(eStat_Mobility, default.HOLOGRAPHIC_DISGUISE_MOBILITY_BONUS);
+	PersistentStatChangeEffect.AddPersistentStatChange(eStat_DetectionModifier, default.HOLOGRAPHIC_DISGUISE_DETECTION_MODIFIER);
+	Template.AddTargetEffect(PersistentStatChangeEffect);
```

Comment:
`PersistentStatChangeEffect.SetDisplayInfo(...)` can be used to show the Detection modifier as a buff/passive status. In combination with X2AbilityTag (see `X2DownloadableContentInfo.uc:AbilityTagExpandHandler`), we could even show the raw value in the passive/buff info. Is this something we want? cc @Xymanek

Reply:
> we could even show the raw value in the passive/buff info

Not sure what you mean

Reply:
I was thinking that the tooltip of the icon in tactical could say "The soldier is wearing Holographic Disguise, detection radius reduced by XX%". But we are dealing with floating point numbers, which are hard to deal with when going from code->localization.

Still, I think a passive icon for the disguise is warranted.

Reply:
Pretty sure I saw `Detection range x0.6` in your F1 at some point

Reply:
I think that this was due to [Show More Buff Details](https://steamcommunity.com/sharedfiles/filedetails/?id=709499969). But right, it could be made to work rudimentarily at least.